examples/userapi/subdomains.ipynb

```diff
@@ -0,0 +1,954 @@
+       "      [0., 0., 0., 0., 1., 1., 1., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 1., 1., 1., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 1., 1., 1., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 25,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "f.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Forward wave propagation example with subdomains\n",
+    "\n",
+    "We will now utilise subdomains in a seismic wave-propagation model. Subdomains are used here to specify different equations in two sections of the domain: in one section we will solve the acoustic wave-equation, and in another the elastic wave-equation. It is useful to review seismic tutorials numbers 5 (staggered acoustic setup) and 6 (elastic setup) to understand the seismic based utilities use below (since here they will simply be 'used').\n",
+    "\n",
+    "We first import all required modules:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from devito import (TimeFunction, VectorTimeFunction, TensorTimeFunction,\n",
+    "                    div, grad, curl, diag)\n",
+    "from examples.seismic import ModelElastic, plot_velocity, TimeAxis, RickerSource, plot_image"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now define the problem dimensions and velocity model:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "extent = (200., 100., 100.) # 200 x 100 x 100 m domain\n",
+    "h = 1.0 # Desired grid spacing\n",
+    "shape = (int(extent[0]/h+1), int(extent[1]/h+1), int(extent[2]/h+1))\n",
+    "\n",
+    "# Model physical parameters:\n",
+    "vp = np.zeros(shape)\n",
+    "qp = np.zeros(shape)\n",
+    "vs = np.zeros(shape)\n",
+    "qs = np.zeros(shape)\n",
+    "rho = np.zeros(shape)\n",
+    "\n",
+    "# Set up three horizontally separated layers:\n",
+    "vp[:,:,:int(0.5*shape[2])+1] = 1.52\n",
+    "vs[:,:,:int(0.5*shape[2])+1] = 0.\n",
+    "rho[:,:,:int(0.5*shape[2])+1] = 1.05\n",
+    "\n",
+    "vp[:,:,int(0.5*shape[2])+1:int(0.5*shape[2])+1+int(4/h)] = 1.6\n",
+    "vs[:,:,int(0.5*shape[2])+1:int(0.5*shape[2])+1+int(4/h)] = 0.4\n",
+    "rho[:,:,int(0.5*shape[2])+1:int(0.5*shape[2])+1+int(4/h)] = 1.3\n",
+    "\n",
+    "vp[:,:,int(0.5*shape[2])+1+int(4/h):] = 2.2\n",
+    "vs[:,:,int(0.5*shape[2])+1+int(4/h):] = 1.2\n",
+    "rho[:,:,int(0.5*shape[2])+1+int(4/h):] = 2.\n",
+    "\n",
+    "origin = (0, 0, 0)\n",
+    "spacing = (h, h, h)"
```

Comment:
`qp = np.zeros(shape)` isn't that for viscoelastic?
Isn't it "easier" to just use a preset?

Reply:
> `qp = np.zeros(shape)` isn't that for viscoelastic?

Yes, remnants I'd forgotten to delete.
> Isn't it "easier" to just use a preset?

I wanted it to be clear to the user how the subdomains line up with the model (which is a bit hidden when using a preset). But I should tidy this up through setting some constants etc.